### PR TITLE
queued online triple generation

### PR DESCRIPTION
**Pull request by mortendahl**
Child of #417 

This PR adds a queued triple source, providing a first step towards separating the online and offline phases using tf.FIFOQueues. The new notebook may be a good place to start to understand what's going on.

### CODE REVIEW

File: tf_encrypted/protocol/pond/triple_sources.py [hunk 1]
```diff
@@ -290,6 +316,7 @@ def _build_queues(self, c0, c1):
 """
 
 
+#pylint: disable=pointless-string-statement
```
Comment (jvmancuso) [inline tf_encrypted/protocol/pond/triple_sources.py]: same as above

File: tf_encrypted/config.py [hunk 2]
```diff
@@ -139,11 +144,18 @@ def add_player(self, name):
   def players(self):
     return self._players
 
-  def get_player(self, name):
-    player = next(
-        (player for player in self._players if player.name == name), None)
+  def get_player(self, name_or_player):
```
Comment (jvmancuso) [inline tf_encrypted/config.py]: should these changes be in a separate commit? or at least in the PR description

Comment (jvmancuso) [reply tf_encrypted/config.py]: they're definitely dependencies of the major code changes, but this is new behavior for a public-facing API so should probably be mentioned

File: tf_encrypted/utils.py [hunk 3]
```diff
@@ -14,3 +14,63 @@ def wrap_in_variables(*tensors):
...
+  - `flatten([x, [y], [[z]]]) => `[x, y, z]`
+  """
+  if isinstance(xs, (list, tuple)):
+    return [y for ys in [flatten(x) for x in xs] for y in ys]
+  return [xs]
+
+
+def reachable_nodes(*nodes):
+  """
+  Find all nodes reachable from `nodes` in the implicit graph.
```
[the remark this answers is not in the document]
Comment (mortendahl) [reply tf_encrypted/utils.py]: the TF graph in which the nodes live

File: tf_encrypted/config.py (hunk 2 above)
Comment (mortendahl) [reply tf_encrypted/config.py]: yeah I thought about it as well, will do that

File: tf_encrypted/protocol/pond/triple_sources.py [hunk 4]
```diff
@@ -188,78 +196,96 @@ class OnlineTripleSource(BaseTripleSource):
...
+      d1 = self._build_triple_store(c1, "1")
 
-    def generate_triples(self, sess, fetches, tag=None):
-        # TODO(Morten) filter which enqueuers to run given fetches
-        sess.run(self.enqueuers, tag=tag)
-"""
+    return d0, d1
 
 
+#pylint: disable=pointless-string-statement
```
[the remark this answers is not in the document]
Comment (mortendahl) [reply tf_encrypted/protocol/pond/triple_sources.py]: you just put it at the end of the text block?
